src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -715,10 +715,16 @@ public IActionResult SendTransaction([FromBody] SendTransactionRequest request)
                     });
                 }
 
-                this.walletManager.ProcessTransaction(transaction, null, null, false);
-
                 this.broadcasterManager.BroadcastTransactionAsync(transaction).GetAwaiter().GetResult();
 
+                var transactionBroadCastEntry = this.walletManager.GetTransactionBroadcastEntry();
```

Comment:
I don't really like to see the walletManager responsible for returning a TrxBroasdcastEntry. What if there are many happening at the same time? Are we guaranteed that they will ever only be 1 to be concerned with?

At least I think we need to have a trx identifier and maybe this method can call  `GetTransaction(uint256 transactionHash)` from the `BroadcasterManagerBase`.

Reply:
Ok, didn't see this was available in BroadcasterManagerBase, thanks.